lib/crawly/spider.ex

```diff
@@ -28,7 +28,7 @@ defmodule Crawly.Spider do
               Crawly.ParsedItem.t()
 
   @callback override_settings() :: Crawly.Settings.t()
-
+  @optional_callbacks parse_item: 1
```

Comment:
I would still keep parse_item as a required callback, as it's going to be the main way of declaring spiders. In our template spider, we can just return empty requests/items from it.